File: backend/risk-analytics-backend/src/main/java/de/charliehillner/riskanalyticsbackend/simulation/Simulation.java
package de.charliehillner.riskanalyticsbackend.simulation;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "simulation")
@Getter
@Setter
@NoArgsConstructor
public class Simulation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "initial_capital")
    private BigDecimal initialCapital;
    @Column(name = "monthly_contribution")
    private BigDecimal monthlyContribution;

    @Column(name = "expected_return")
    private double expectedReturn;
    @Column(name = "volatility")
    private double volatility;

    @Column(name = "years")
    private int years;
    @Column(name = "num_runs")
    private int numRuns;

    @Column(name = "mean_final_value")
    private BigDecimal meanFinalValue;
    @Column(name = "median_final_value")
    private BigDecimal medianFinalValue;
    @Column(name = "percentile_5")
    private BigDecimal percentile5;
    @Column(name = "percentile_95")
    private BigDecimal percentile95;

    @Column(name = "loss_probability")
    private double lossProbability;
}

File: frontend/src/types/simulation.ts
export type SimulationRequest = {
  initialCapital: number;
  monthlyContribution: number;
  expectedReturn: number;
  volatility: number;
  years: number;
  numRuns: number;
};

export type SimulationResponse = {
  id: number;
  message: string;
  meanFinalValue: number;
  medianFinalValue: number;
  percentile5: number;
  percentile95: number;
  lossProbability: number;
  finalValues: number[];
  valueAtRisk95: number;
  conditionalValueAtRisk95: number;
};

export type SimulationHistoryItem = {
  id: number;
  createdAt: string;
  initialCapital: number;
  monthlyContribution: number;
  expectedReturn: number;
  volatility: number;
  years: number;
  numRuns: number;
  meanFinalValue: number;
  medianFinalValue: number;
  percentile5: number;
  percentile95: number;
  lossProbability: number;
  valueAtRisk95: number;
  conditionalValueAtRisk95: number;
};